core/store/src/flat/store_helper.rs

```diff
@@ -197,17 +222,18 @@ pub fn remove_flat_storage_creation_status(store_update: &mut StoreUpdate, shard
 }
 
 /// Iterate over flat storage entries for a given shard.
-/// It reads data only from the 'main' column - which represents the state as of final head.k
+/// It reads data only from the 'main' column - which represents the state as of final head.
 ///
 /// WARNING: flat storage keeps changing, so the results might be inconsistent, unless you're running
 /// this method on the shapshot of the data.
+// TODO(#8676): Support non-trivial ranges.
 pub fn iter_flat_state_entries<'a>(
```

Comment:
it looks like we also need to chage this method to accept `shard_uid` instead of `shard_id` and do a proper prefix-read? or do you please to do that as a separate PR?

Reply:
Yeah, I would do it separately. Maybe even outside MVP, because we don't use this logic much.